### ISSUE TITLE
serializer_class must be specified even when using @view_config(request_serializer=...)

### ISSUE THREAD

**Issue by bikeshedder** [limdauto/drf_openapi#57]
* DRF OpenAPI version: 1.0 (installed from pypi)
* Python version: 3.6
* Operating System: Fedora 26

### Description

When adding a `view_config` decorator to a `APIView` method the `request_serializer` parameter is ignored when viewing the schema. The only way to get the request body rendered is by adding a `serializer_class` to the `APIView`

### What I Did

```python
from django.conf.urls import url
from drf_openapi.utils import view_config
from rest_framework import status
from rest_framework.views import APIView
from rest_framework.parsers import FormParser, MultiPartParser
from rest_framework.response import Response

from . import serializers

class UploadView(APIView):
    serializer_class = serializers.UploadRequestSerializer # <---- should not be needed
    parser_classes = (FormParser, MultiPartParser)

    @view_config(
        request_serializer=serializers.UploadRequestSerializer,
        response_serializer=serializers.UploadResponseSerializer)
    def post(self, request, version, format=None):
        req = self.request_serializer(data=request.data)
        req.is_valid(raise_exception=True)
        upload = req.save()
        res = self.response_serializer(upload)
        return Response(res.data, status=status.HTTP_201_CREATED)
```

When leaving out the `serializer_class` no request body is shown in the schema. This is obviously a limitation when `post` and `put` do not use the same request serializer.

As the example in the documentation does not use the `serializer_class` I think this must be a bug: https://drf-openapi.readthedocs.io/en/latest/usage.html#add-schema-to-a-view-method

**Comment by limdauto** [comment]
This is odd. Thanks. Gonna investigate.

**Comment by bikeshedder** [comment]
I'm having more troubles with this than initially thought. I tried using the `@view_config` decorator in a ViewSet and had to realize that I need to specify a different `request_serializer` for the list and create method. Therefore the `serializer_class` doesn't help much.

I tried digging into it...

https://github.com/limdauto/drf_openapi/blob/9813af66ff2e71f34bfb32643649b2140694a63e/drf_openapi/entities.py#L153

https://github.com/limdauto/drf_openapi/blob/9813af66ff2e71f34bfb32643649b2140694a63e/drf_openapi/entities.py#L278

 I found that `fields` is only populated by fields from the `serializer_class` ignoring the `method.request_serializer`. If I understand the code correctly this part of the code should be patched.

**Comment by bikeshedder** [comment]
I just realized that I messed up the `get_serializer_class` method in my pull request. It needs to use `serializer_class` and "get_serializer_class" from the `view` and `response_serializer` from the `view_func`. Right now it tried to load everything from the `view_func` which is just wrong.

**Comment by limdauto** [comment]
I haven't really merged / carefully reviewed your PR so please feel free to amend it.

**Comment by flyte** [comment]
Just a friendly bump on this PR - it's a bit of a deal breaker for me to use this project.

@bikeshedder are you done making your updates?

**Comment by limdauto** [comment]
This bug should be fixed with in 1.2.0. /cc @werwty

**Comment by bikeshedder** [comment]
Sorry for not responding earlier. I recently moved away from DRF and DRF OpenAPI and try my luck with a schema first attempt using [Connexion](https://github.com/zalando/connexion). I'm glad that this issue is fixed in the latest release of DRF OpenAPI.
